### PR TITLE
chore: Clean up some duplicated code.

### PR DESCRIPTION
**Pull request by He-Pin**
Motivation：
I found these code is a little duplicated.

Result:
Simpler code.

### CODE REVIEW

File: stream/src/main/scala/org/apache/pekko/stream/impl/fusing/Ops.scala
Comment (pjfanning) [inline stream/src/main/scala/org/apache/pekko/stream/impl/fusing/Ops.scala:272]: can you explain the change? - does `pull(in)` automatically check `!hasBeenPulled(in)`?

**Review by pjfanning — CHANGES_REQUESTED** [review]
change needs better explanation - you should not expect reviewers to read all the other code for themselves

File: stream/src/main/scala/org/apache/pekko/stream/impl/fusing/Ops.scala
Comment (He-Pin) [reply stream/src/main/scala/org/apache/pekko/stream/impl/fusing/Ops.scala:272]: There is no need to check the `(!hasBeenPulled(in))` , because there is no additional `pull(in)`.  these method was copied from the old code, where, an exception maybe throw during the `preStart` but the upstream has been pulled. but here, as the `collect` operator never do some pulling or any other things which can throws after a `pull(in)`, so I think it would be better to just remove those duplicated check.

That's why I remove it.

**Review by pjfanning — APPROVED** [review]
lgtm - but it would be good if someone else reviews this too

File: stream/src/main/scala/org/apache/pekko/stream/impl/fusing/Ops.scala
Comment (mdedetrich) [inline stream/src/main/scala/org/apache/pekko/stream/impl/fusing/Ops.scala:271]: Can we add a comment here explaining why `!hasBeenPulled(in)` is not necessary.

**Review by mdedetrich — CHANGES_REQUESTED** [review]
While I trust that the `!hasBeenPulled(in)` is not necessary it would be good to add a comment explaining why for future people reading the code.

Would also be good to add a unit test confirming this if its quick to do so.

Ill approve once comment is added

File: stream/src/main/scala/org/apache/pekko/stream/impl/fusing/Ops.scala
Comment (He-Pin) [reply stream/src/main/scala/org/apache/pekko/stream/impl/fusing/Ops.scala:271]: Why? the Map and Filter don't have this either.

I would like to add that for in `MapConcat` operator which explain why a `!hasBeenPulled(in)` test is needed.

I think the origin reason about why this `!hasBeenPulled(in)` test is because that method is a abstract class and so that exception can happen when onPush/onPull, so it have to be guard for extending, and reduce the sub class' burden. that's not the case here.

**Comment by He-Pin** [comment]
@mdedetrich @pjfanning I can not push anycode now, GFW blocked my network.

File: stream/src/main/scala/org/apache/pekko/stream/impl/fusing/Ops.scala
Comment (mdedetrich) [reply stream/src/main/scala/org/apache/pekko/stream/impl/fusing/Ops.scala:271]: > Why? the Map and Filter don't have this either.

We should add the comments there as well then. Its more that when other people that are not so familiar with Akka/Pekko read the code they get a better understanding of why these exceptions are there.

Akka's problem historically is that a lot of the contributions were done within Lightbend and so there is a lot of inhouse knowledge that wasn't visible to outside contributors.

Comment (He-Pin) [reply stream/src/main/scala/org/apache/pekko/stream/impl/fusing/Ops.scala:271]: I can't push any code because gfw blocked me, Maybe I can use a web ide?
How about we do that in a separate pr or you take care about it?

Comment (He-Pin) [reply stream/src/main/scala/org/apache/pekko/stream/impl/fusing/Ops.scala:271]: https://pekko.apache.org/docs/pekko/current/stream/stream-customize.html
It's detailly described here.

Comment (He-Pin) [reply stream/src/main/scala/org/apache/pekko/stream/impl/fusing/Ops.scala:271]: @mdedetrich Please unlock this, I'm not pay for everything, thanks, I just a volunteer, and this kind of information is well covered in the doc, this is code, not a book.

Comment (mdedetrich) [reply stream/src/main/scala/org/apache/pekko/stream/impl/fusing/Ops.scala:271]: @He-Pin I was away from laptop, ill just add a commit with comment onto your branch and then merge it.

File: stream/src/main/scala/org/apache/pekko/stream/impl/fusing/Ops.scala
Comment (mdedetrich) [inline stream/src/main/scala/org/apache/pekko/stream/impl/fusing/Ops.scala:271]: @He-Pin Is this comment correct?

Comment (He-Pin) [reply stream/src/main/scala/org/apache/pekko/stream/impl/fusing/Ops.scala:271]: Is true, but where this comments come from?

Comment (mdedetrich) [reply stream/src/main/scala/org/apache/pekko/stream/impl/fusing/Ops.scala:271]: As a maintainer of Pekko, I have permission to add commits onto branches of other people's PR. This is my commit that I added now.

I will approve PR and squash + merge it, thanks for work!

**Review by mdedetrich — APPROVED** [review]
lgtm

File: stream/src/main/scala/org/apache/pekko/stream/impl/fusing/Ops.scala
Comment (He-Pin) [reply stream/src/main/scala/org/apache/pekko/stream/impl/fusing/Ops.scala:271]: I see, hah, The chinese networking to github is very unstable, that's why I hate changing and changing, sometime I can't pulling pull too, both via ssh and https

**Pull request merged**
